internal/reconciler/telemetry/reconciler.go

```diff
@@ -105,6 +105,10 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return ctrl.Result{}, fmt.Errorf("failed to clean up old network policies: %w", err)
 	}
 
+	if err := r.cleanUpOpenCensus(ctx); err != nil {
```

Comment:
maybe we can be more specific and rename it to `cleanUpOpenCensusService`

Reply:
Do we want to fail the reconciliation if cleanup fails? Maybe just log and swallow the error?

Reply:
@shorim What will you say?